FileSystem/Util.js

```diff
@@ -38,3 +38,96 @@ function deleteFile(name) {
     DriveApp.removeFile(fileList.next());
   }
 }
+
+/**
+ * Computes the Drive path for a file/folder using their Id
+ * @param {String} id The drive destination Id whose path needs to be computed
+ * @return {String} drivePath The corresponding drive path
+ */
+function getFullDrivePath(id, isFile) {
+  var current;
+
+  if (isFile) {
+    current = DriveApp.getFileById(id);
+  } else {
+    current = DriveApp.getFolderById(id);
+  }
+
+  var folders = [];
+  var parentIterator = current.getParents();
+
+  while (parentIterator.hasNext()) {
+    var parent = parentIterator.next();
+    folders.push(parent.getName());
+    parentIterator = parent.getParents();
+  }
+
+  var drivePath = folders.reverse().join('/') + '/' + current.getName();
+  return drivePath;
+}
+
+/**
+ * File System Type Enumeration
+ */
+var FileSystemType = {
+  WINDOWS: 'windows',
+  UNIX: 'unix',
+};
+
+/**
+ * @todo Write tests for validation
+ * @body Tests will be written in the directory API module as these are more closely related to that module.
+ */
+// Regex expressions to validate absolute paths
+var windowsPathRegExp = /^[\w]\:(\\|(\\[^<>\\/:"\|\?\*]+)+)\\?$/;
+var unixPathRegExp = /^(\/[^<>\\/:"\|\?\*]+)*\/?$/;
+
+/**
+ * Helper function to validate path
+ * @param {string} path File or Directory path
+ * @return {boolean} true if path is valid
+ */
+function isValidAbsolutePath(path) {
+  if (windowsPathRegExp.test(path)) {
+    return true;
+  } else if (unixPathRegExp.test(path)) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Helper function to obtain path type
+ * @param {string} path File or directory path
+ * @return {string} File System type enumeration
+ */
+function getFileSystemType(path) {
+  if (windowsPathRegExp.test(path)) {
+    return FileSystemType.WINDOWS;
+  } else if (unixPathRegExp.test(path)) {
+    return FileSystemType.UNIX;
+  }
+  throw new Error('Unknown FileSystem');
+}
+
+/**
+ * Helper function to remove trailing file separator
+ * @param {string} path File or directory path
```

Comment:
Please mention that you are referring to drive path here.

Reply:
Fixed, it is referring to the local file path

Reply:
It will be better if you can use variableNames accordingly.
Signature sanitizePath(path) does not say much whether you are sanitising the local path/drive path.

Reply:
Fixed